lib/src/utils/screen_loading/screen_loading_trace.dart

```diff
@@ -0,0 +1,46 @@
+class ScreenLoadingTrace {
+  ScreenLoadingTrace(
+    this.screenName, {
+    required this.startTimeInMicroseconds,
+    this.endTimeInMicroseconds,
+    this.duration,
+  });
+
+  final String screenName;
+  int startTimeInMicroseconds;
+  int? endTimeInMicroseconds;
+  int? duration;
+
+  ScreenLoadingTrace copyWith({
+    String? screenName,
+    int? startTimeInMicroseconds,
+    int? endTimeInMicroseconds,
+    int? duration,
+  }) {
+    return ScreenLoadingTrace(
+      screenName ?? this.screenName,
+      startTimeInMicroseconds:
+          startTimeInMicroseconds ?? this.startTimeInMicroseconds,
+      endTimeInMicroseconds:
+          endTimeInMicroseconds ?? this.endTimeInMicroseconds,
+      duration: duration ?? this.duration,
+    );
+  }
+
+
+  @override
+  bool operator ==(Object other) =>
+      identical(this, other) ||
+      other is ScreenLoadingTrace &&
+          runtimeType == other.runtimeType &&
+          screenName == other.screenName &&
+          startTimeInMicroseconds == other.startTimeInMicroseconds;
```

Comment:
Question: Do we need to overload the `==` operator for this class? I'd love to hear your thoughts 🙏🏼